go/host/db/hostdb.go

```diff
@@ -38,25 +41,42 @@ type DB struct {
 	blockReads  gethmetrics.Gauge
 }
 
+func (db *DB) Start() error {
+	return nil
+}
+
 // Stop is especially important for graceful shutdown of LevelDB as it may flush data to disk that is currently in cache
-func (db *DB) Stop() error {
+func (db *DB) Stop() {
 	db.logger.Info("Closing the host DB.")
 	err := db.kvStore.Close()
 	if err != nil {
-		return err
+		db.logger.Error("Error closing the host DB.", log.ErrKey, err)
 	}
-	return nil
 }
 
-func CreateDBFromConfig(cfg *config.HostConfig, regMetrics gethmetrics.Registry, logger gethlog.Logger) (*DB, error) {
+func (db *DB) HealthStatus() hostcommon.HealthStatus {
+	// always healthy for now, satisfies Service interface
+	return &hostcommon.BasicErrHealthStatus{ErrMsg: ""}
+}
+
+// InitialiseMetrics registers the host DB metrics with the given registry (this is called by the metrics service when it starts)
+func (db *DB) InitialiseMetrics(registry gethmetrics.Registry) {
```

Comment:
If this happens everytime the DB is created, why make a new method ?

Can it be that the problem is that the inmem db is using a struct direct initialization instead of a function ? I mean it would be easy to do a check for nil registry right ?

Reply:
Perhaps if this was controlled by the host container then it would be easy to manage the dependecies. Like the registry would be created and then when the db is created with the registry injection in it.